sympy/ntheory/generate.py

```diff
@@ -244,11 +247,35 @@ def primepi(n):
         prime : Return the nth prime
     """
     n = int(n)
-    if n < 2:
-        return 0
-    else:
-        return sieve.search(n)[0]
-
+    if n<2:
+		return 0	
+	lim = int(n**0.5) #lim = floor(sqrt(n))
```

Comment:
This will lose precision if n is big enough. If that matters, you should use `floor(sqrt(n))` where floor and sqrt from SymPy. However, n has to be quite big for this to happen, so I don't know if this algorithm can ever return a value for those numbers anyway.